osu.Game.Tests/Visual/Playlists/TestScenePlaylistsResultsScreen.cs

```diff
@@ -53,12 +53,23 @@ public class TestScenePlaylistsResultsScreen : ScreenTestScene
             userScore.Statistics = new Dictionary<HitResult, int>();
 
             bindHandler();
-
-            // beatmap is required to be an actual beatmap so the scores can get their scores correctly calculated for standardised scoring.
-            // else the tests that rely on ordering will fall over.
-            Beatmap.Value = CreateWorkingBeatmap(Ruleset.Value);
         });
 
+        [SetUpSteps]
+        public override void SetUpSteps()
+        {
+            base.SetUpSteps();
+
+            // Existing test instances of the results screen hold a leased bindable of the beatmap,
+            // so wait for those screens to be cleaned up by the base SetUpSteps before re-assigning
```

Comment:
My main question here is: if this ordering is a problem, then why isn't everything in `SetUp()` moved to `SetUpSteps()`? Which is to say, that if there is one issue that is caused by a stale results screen changing internal state of the test, then aren't *all* pieces of state in this test susceptible to that, rather than just the leased beatmap bindable?

Reply:
Ideally the process of exiting all existing screens should happen before `SetUp` and not after but that seems like a much larger piece of work. Should I open an issue for it?
For the purposes of this test, this local fix seems fine for now.

Reply:
I'm not questioning the validity of the local fix or asking to address every single instance of this misordering. I'm fine with a local fix, but I'm questioning whether this fix going far enough. It would not take much work to move everything out of `SetUp()` into `SetUpSteps()`, and it would make sure that other pieces of state in this test cannot suffer from the same bug.

Reply:
Oh I see what you mean. I will move the the rest of the setup down too just in case.